File: ws_attempt/data_ws/process_ticket_sales.py
import pandas as pd
import numpy as np

# Read the ticket sales data
ticket_data = pd.read_csv('TicketSales.csv')

# Convert Transaction Date to datetime format
ticket_data['Transaction_Date'] = pd.to_datetime(ticket_data['Transaction Date'], errors='coerce')

# Aggregate ticket sales by date
daily_ticket_sales = (ticket_data
    .dropna(subset=['Transaction_Date'])
    .groupby(ticket_data['Transaction_Date'].dt.date)
    .size()
    .reset_index(name='ticket_sales')
    .rename(columns={'Transaction_Date': 'date_id'})
)

# Convert date_id to string format for compatibility
daily_ticket_sales['date_id'] = daily_ticket_sales['date_id'].astype(str)

# Save the aggregated data
daily_ticket_sales.to_csv('daily_ticket_sales.csv', index=False)

print("Ticket sales data processed and saved to daily_ticket_sales.csv")

File: ws_attempt/data_ws/process_ticket_sales.R
library(dplyr)
library(lubridate)

# Read the ticket sales data
ticket_data <- read.csv("TicketSales.csv")

# Convert Transaction Date to Date format
ticket_data$Transaction_Date <- as.Date(ticket_data$Transaction_Date)

# Aggregate ticket sales by date
daily_ticket_sales <- ticket_data %>%
  filter(!is.na(Transaction_Date)) %>%
  group_by(Transaction_Date) %>%
  summarise(ticket_sales = n()) %>%
  rename(date_id = Transaction_Date)

# Save the aggregated data
write.csv(daily_ticket_sales, "daily_ticket_sales.csv", row.names = FALSE)

print("Ticket sales data processed and saved to daily_ticket_sales.csv")